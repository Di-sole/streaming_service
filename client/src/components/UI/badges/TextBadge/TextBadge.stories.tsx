import { Meta, StoryObj } from '@storybook/react';
import './TextBadge.module.scss';
import Badge from './TextBadge';

const meta: Meta<typeof Badge> = {
    title: 'Badges/TextBadge',
    component: Badge,
};

export default meta;

type Story = StoryObj<typeof Badge>;

export const TextBadge: Story = {
    args: {
        text: 'Текст',
    },
};
