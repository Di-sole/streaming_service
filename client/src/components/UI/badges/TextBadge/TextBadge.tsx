import style from './TextBadge.module.scss';

interface TextBadgeProps {
    text: string;
}

const TextBadge = ({ text }: TextBadgeProps) => {
    return (
        <div className={style.badge} data-testid={'textBadge'}>
            {text}
        </div>
    );
};

export default TextBadge;
