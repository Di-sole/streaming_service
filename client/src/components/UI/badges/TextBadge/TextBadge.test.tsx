import { render, screen } from '@testing-library/react';
import TextBadge from './TextBadge';

describe('TEXTBADGE SNAPSHOTS TESTS', () => {
    test('TextBadge should not be changed', () => {
        render(<TextBadge text="text" />);

        const id = screen.getByTestId('textBadge');
        expect(id).toMatchSnapshot();
    });
});
