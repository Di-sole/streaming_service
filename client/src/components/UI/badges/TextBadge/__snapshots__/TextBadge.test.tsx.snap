// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEXTBADGE SNAPSHOTS TESTS TextBadge should not be changed 1`] = `
<div
  class="badge"
  data-testid="textBadge"
>
  text
</div>
`;
